physical/raft/raft.go

```diff
@@ -1162,18 +1185,28 @@ func (l *RaftLock) monitorLeadership(stopCh <-chan struct{}, leaderNotifyCh <-ch
 // Lock blocks until we become leader or are shutdown. It returns a channel that
 // is closed when we detect a loss of leadership.
 func (l *RaftLock) Lock(stopCh <-chan struct{}) (<-chan struct{}, error) {
-	l.b.l.RLock()
+	raftInitialized := l.b.Initialized()
 
-	// Cache the notifyCh locally
-	leaderNotifyCh := l.b.raftNotifyCh
+	// If not initialized, block until it is
+	if !raftInitialized {
+		select {
+		case <-l.b.raftInitCh:
+		case <-stopCh:
+			return nil, nil
+		}
+
+	}
 
-	// TODO: Remove when Raft can server as the ha_storage backend. The internal
-	// raft pointer should not be nil here, but the nil check is a guard against
-	// https://github.com/hashicorp/vault/issues/8206
+	l.b.l.RLock()
+
+	// Ensure that we still have a raft instance after grabbing the read lock
 	if l.b.raft == nil {
 		return nil, errors.New("attempted to grab a lock on a nil raft backend")
```

Comment:
it's safe to return here without unlocking the read lock, correct?

Reply:
Nice catch!